src/ucp/core/ucp_worker.c

```diff
@@ -427,98 +427,19 @@ void ucp_worker_signal_internal(ucp_worker_h worker)
-                                          UCS_CALLBACKQ_FLAG_ONESHOT, &prog_id);
-        return 0;
-    }
-
-    UCS_ASYNC_BLOCK(&worker->async);
-
-    ucs_debug("ep %p: handle error: %s", ucp_ep, ucs_status_string(status));
-
-    ucs_assert(ucp_ep->flags & UCP_EP_FLAG_FAILED);
-
-    /* the EP can be closed from last completion callback */
-    ucp_ep_discard_lanes(ucp_ep, status);
-    ucp_ep_reqs_purge(ucp_ep, status);
-    ucp_stream_ep_cleanup(ucp_ep);
-
-    if (ucp_ep->flags & UCP_EP_FLAG_USED) {
-        if (ucp_ep->flags & UCP_EP_FLAG_CLOSE_REQ_VALID) {
-            ucs_assert(ucp_ep->flags & UCP_EP_FLAG_CLOSED);
-            /* Promote close operation to CANCEL in case of transport error,
-             * since the disconnect event may never arrive. */
-            close_req                        = ucp_ep_ext_control(ucp_ep)->
-                                               close_req.req;
-            close_req->send.flush.uct_flags |= UCT_FLUSH_FLAG_CANCEL;
-            ucp_ep_local_disconnect_progress(close_req);
-        } else {
-            ucp_ep_invoke_err_cb(ucp_ep, status);
-        }
-    } else if (ucp_ep->flags & (UCP_EP_FLAG_INTERNAL | UCP_EP_FLAG_CLOSED)) {
-        /* No additional actions are required, this is already closed EP or
-         * an internal one for sending WIREUP/EP_REMOVED messsage to a peer.
-         * So, close operation was already scheduled, this EP will be deleted
-         * after all lanes will be discarded successfully */
-        ucs_debug("ep %p: detected peer failure on internal endpoint", ucp_ep);
-    } else {
-        ucs_debug("ep %p: destroy endpoint which is not exposed to a user due"
-                  " to peer failure", ucp_ep);
-        ucp_ep_disconnected(ucp_ep, 1);
-    }
-
-    ucs_free(err_handle_arg);
-    UCS_ASYNC_UNBLOCK(&worker->async);
-    return 1;
-}
-
-int ucp_worker_err_handle_remove_filter(const ucs_callbackq_elem_t *elem,
-                                        void *arg)
-{
-    ucp_worker_err_handle_arg_t *err_handle_arg = elem->arg;
-
-    if ((elem->cb == ucp_worker_iface_err_handle_progress) &&
-        (err_handle_arg->ucp_ep == arg)) {
-        /* release err handling argument to avoid memory leak */
-        ucs_free(err_handle_arg);
-        return 1;
-    }
-
-    return 0;
-}
-
 /*
  * Caller must acquire lock
  */
 ucs_status_t ucp_worker_set_ep_failed(ucp_worker_h worker, ucp_ep_h ucp_ep,
                                       uct_ep_h uct_ep, ucp_lane_index_t lane,
                                       ucs_status_t status)
 {
-    uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
-    ucs_status_t ret_status    = UCS_OK;
-    char lane_info_str[64];
-    ucp_rsc_index_t rsc_index;
-    uct_tl_resource_desc_t *tl_rsc;
-    ucp_worker_err_handle_arg_t *err_handle_arg;
+    ucs_status_t ret_status = UCS_OK;
     ucs_log_level_t log_level;
+    ucp_request_t *close_req;
+    UCS_STRING_BUFFER_ONSTACK(lane_info_strb, 64);
```

Comment:
this is an initialized var, it should be in the beginning

Reply:
done